charts/apps/custom-app/chart/templates/deployment.yaml

```diff
@@ -42,7 +48,11 @@ spec:
         command:
           - /bin/sh
           - -c
-          - cp -r {{ $.Values.appconfig.path }}/* /tmp
+          {{- if $.Values.admin }}
+          - cp -r -n {{ $.Values.appconfig.path }}/* /tmp
+          {{- else }}
+          - cp -r -n /home/{{ $.Values.appconfig.path }}/* /tmp
+          {{- end }}
```

Comment:
I suggest that we think more about this statement.
It should do the following, i think:
1. Check if `/home/{{ $.Values.appconfig.path }}` exists
2. Check if something similar to `/home/{{ $.Values.appconfig.path }}`  exists. If the user types `FolderName`, but the folder is called `folderName`, it should work.
3. Check permissions?

and, what should happen if it fails? Should we proceed anyways, without copying the data? Return failed container? Perhaps we could create a file in the PVC saying something like "Could not copy due to <reason>" or something like this? 

Maybe something like this

```bash
command: ["/bin/sh", "-c"]
    args: 
      - |
        if [ -d "/home/{{ $.Values.appconfig.path }}" ]; then
          # Step 1: Check if /home/{{ $.Values.appconfig.path }} exists and is directory
          
          if [ -r "/home/{{ $.Values.appconfig.path }}" ]; then
            # Step 1b: If it exists, then check if exists and is readable
            
            # Proceed to copy
            cp -r -n /home/{{ $.Values.appconfig.path }}/* /tmp
          else
            # Step 1c: If not permitted, return error exit code
            echo "Error: No read permissions for /home/{{ $.Values.appconfig.path }}"
            exit 1
          fi
          
        else
          # Step 2: If the path does not exist, find the first similar path using regex
          similar_path=$(find /home -type d -regex ".*/{{ $.Values.appconfig.path }}.*" -print -quit)
          
          if [ -n "$similar_path" ]; then
            # Step 2b: If found, then copy
            cp -r -n "$similar_path"/* /tmp
          else
            # Step 2c: If not found, return error exit code
            echo "Error: Path not found - /home/{{ $.Values.appconfig.path }}"
            exit 1
          fi
        fi

```

Reply:
Thanks for the feedback and providing these examples Viktor. I definitely agree with point 1. 

Point 2 and 3 also make sense but these are also things that the logs can show as well. If there are permission issues, the logs will clearly say Permission denied. We now have logs for both the custom app container and the copy to pvc init container. I would also say that it might be good to just let the user know that the folder is not found (this should also be shown in the logs). The user can then correct it in the path.

Reply:
Ok! I agree that the logs will show it, but we should aim for making stuff flexible and work eventhough the user did some simple mistake, like spelling mistake or similar. We can work on these kind of features later aswell.

Reply:
Agreed. I will add this to the backlog!